ERCS/erc-simulate-calls.md

```diff
@@ -0,0 +1,227 @@
+  // Estimated total gas used by the calls.
+  gasUsed: `0x${string}`;
+  // Logs emitted by the calls.
+  logs: {
+    address: `0x${string}`;
+    data: `0x${string}`;
+    topics: `0x${string}`[];
+  }[];
+  // ERC-5792 Capabilities results.
+  capabilities?: Record<string, any>;
+  // Error that occurred during simulation.
+  error?: {
+    message: string;
+    data?: `0x${string}`;
+    details?: string;
+  },
+  // Status code of the simulation. 
+  status: number;
+}
+```
+
+#### Example
+
+##### Successful Simulation
+
+```ts
+const response = await provider.request({
+  method: 'wallet_simulateCalls',
+  params: [{
+    calls: [{
+      to: '0xcafebabecafebabecafebabecafebabecafebabe',
+      data: '0xdeadbeef',
+    }],
+    chainId: '0x1',
+    from: '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+    version: '1',
+  }],
+});
+
+console.log(response);
+/**
+ * {
+ *  chainId: '0x1',
+ *  gasUsed: '0xe208',
+ *  logs: [{
+ *    address: '0xcafebabecafebabecafebabecafebabecafebabe',
+ *    data: '0x0000000000000000000000000000000000000000000000000000000000069420',
+ *    topics: [
+ *      '0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef',
+ *      '0x000000000000000000000000deadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+ *      '0x000000000000000000000000cafebabecafebabecafebabecafebabecafebabe',
+ *    ],
+ *  }],
+ *  status: 200,
+ * }
+ */
+```
+
+##### Failed Simulation (Onchain Failure)
+
+```ts
+const response = await provider.request({
+  method: 'wallet_simulateCalls',
+  params: [{
+    calls: [{
+      to: '0xba11ba11ba11ba11ba11ba11ba11ba11ba11ba11',
+      data: '0xbabebabe0000000000000000000000000000000000000000000000000000000000000001',
+    }],
+    chainId: '0x1',
+    from: '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+    version: '1',
+  }],
+});
+
+console.log(response);
+/**
+ * {
+ *  chainId: '0x1',
+ *  gasUsed: '0xe208',
+ *  error: {
+ *    message: 'Token ID is taken.',
+ *    details: 'execution reverted: TokenIdUnavailable()',
+ *    data: '0x08c379a0',
+ *  },
+ *  status: 500,
+ * }
+ */
+```
+
+##### Failed Simulation (Offchain Failure)
+
+```ts
+const response = await provider.request({
+  method: 'wallet_simulateCalls',
+  params: [{
+    calls: [{
+      to: '0xba11ba11ba11ba11ba11ba11ba11ba11ba11ba11',
+      data: '0xbabebabe0000000000000000000000000000000000000000000000000000000000000001',
+    }],
+    chainId: '0x1',
+    from: '0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef',
+    version: '1',
+  }],
+});
+
+console.log(response);
+/**
+ * {
+ *  chainId: '0x1',
+ *  gasUsed: '0xe208',
+ *  error: {
+ *    message: 'Paymaster balance is insufficient.',
+ *    details: 'useroperation reverted during simulation with reason: aa26',
+ *  },
+ *  status: 400,
+ * }
+ */
+```
+
+### `traceAssetChanges` Capability
+
+Capability to trace ERC20 & Native asset changes. Needs discussion.
+
+### `traceMalicious` Capability
+
+Capability to trace malicious calls. Needs discussion.
+
```

Comment:
Maybe also a separate `detectMaliciousAddresses` capability? It seems functionally separate from `traceMalicious`; it'd be used to flag i.e. token transfers to known scam addresses?